docs/develop/understanding-the-javascript-api-for-office.md

```diff
@@ -28,31 +28,91 @@ For more details around the Office.js CDN, including how versioning and backward
+            console.log("Sorry, this add-in only works with newer versions of Excel.");
+        }
+})();
+```
+
+If you are using additional JavaScript frameworks that include their own initialization handler or tests, these should be placed within the response to `Office.onReady()`. For example, [JQuery's](https://jquery.com) `$(document).ready()` function would be referenced as follows:
+
+```js
+Office.onReady(function() {
     // Office is ready
-    $(document).ready(function () {        
+    $(document).ready(function () {
         // The document is ready
     });
-  };
+});
 ```
 
-All pages within an Office Add-ins are required to assign an event handler to the initialize event, **Office.initialize**.
-If you fail to assign an event handler, your add-in may raise an error when it starts. Also, if a user attempts to use your add-in with an Office Online web client, such as Excel Online, PowerPoint Online, or Outlook Web App, it will fail to run. If you don't need any initialization code, then the body of the function you assign to **Office.initialize** can be empty, as it is in the first example above.
+### Initialize with Office.initialize
+
+An initialize event fires when the Office.js library is fully loaded and ready for user interaction. You can assign a handler to `Office.initialize` that implements your initialization logic. The following is an example that checks to see that the user's version of Excel supports all the APIs that the add-in might call.
+
+```js
+Office.initialize = function () {
+    if (!Office.context.requirements.isSetSupported('ExcelApi', '1.7')) {
+        console.log("Sorry, this add-in only works with newer versions of Excel.");
+    }
+};
+```
 
-For more detail about the sequence of events when an add-in is initialized, see [Loading the DOM and runtime environment](loading-the-dom-and-runtime-environment.md).
+If you are using additional JavaScript frameworks that include their own initialization handler or tests, these should be placed within the `Office.initialize` event. For example, [JQuery's](https://jquery.com) `$(document).ready()` function would be referenced as follows:
+
+```js
+Office.initialize = function () {
+    // Office is ready
+    $(document).ready(function () {
+        // The document is ready
+    });
+  };
+```
 
-#### Initialization reason
-For task pane and content add-ins, Office.initialize provides an additional _reason_ parameter. This parameter can be used to determine how an add-in was added to the current document. You can use this to provide different logic for when an add-in is first inserted versus when it already existed within the document. 
+For task pane and content add-ins, `Office.initialize` provides an additional _reason_ parameter. This parameter can be used to determine how an add-in was added to the current document. You can use this to provide different logic for when an add-in is first inserted versus when it already existed within the document.
```

Comment:
Nit: "This parameter can be used to determine" -> "This parameter is used to determine" (or even "This parameter determines")